levelZeroLib/src/levelZeroPowerMonitor.cpp

```diff
@@ -0,0 +1,274 @@
+
+    jsize length = env->GetArrayLength(jSysmanDevices);
+    std::vector<zes_device_handle_t> allSysmanDevices(length);
+    jlong* elements = env->GetLongArrayElements(jSysmanDevices, nullptr);
+
+    for (jsize i = 0; i < length; ++i) {
+        allSysmanDevices[i] = reinterpret_cast<zes_device_handle_t>(elements[i]);
+    }
+
+    env->ReleaseLongArrayElements(jSysmanDevices, elements, 0);
+
+    if (allSysmanDevices.empty()) {
+        return -1; // No Sysman devices found
+    }
+
+    std::vector<zes_device_handle_t>  sysmanDevicesToQuery;
+    for (auto& sysmanDevice : allSysmanDevices) {
+        if (!IsIntegratedGPU(sysmanDevice)) {
+            sysmanDevicesToQuery.push_back(sysmanDevice);
+        }
+    }
+   
+    if (sysmanDevicesToQuery.empty()) {
+        return -2; // All iGPUs - Cannot calculate power consumption
+    }
+    else{
+        return 0; //Success
+    }
+
+}
+
+/*
+ * Class:     uk_ac_manchester_tornado_drivers_spirv_levelzero_LevelZeroPowerMonitor
+ * Method:    getSysmanDevicesToQuery
+ * Signature: ([J)[J
+ */
+JNIEXPORT jlongArray JNICALL Java_uk_ac_manchester_tornado_drivers_spirv_levelzero_LevelZeroPowerMonitor_getSysmanDevicesToQuery
+  (JNIEnv* env, jobject obj, jlongArray jSysmanDevices){
+
+    // Convert jlongArray to a C++ array
+    jsize length = env->GetArrayLength(jSysmanDevices);
+    jlong* sysmanDevices = env->GetLongArrayElements(jSysmanDevices, NULL);
+
+    std::vector<jlong> sysmanDevicesToQuery;
+
+    for (jsize i = 0; i < length; ++i) {
+        jlong sysmanDevice = sysmanDevices[i];
+        
+        // Cast jlong to zes_device_handle_t
+        zes_device_handle_t hSysmanDevice = reinterpret_cast<zes_device_handle_t>(sysmanDevice);
+        if (!IsIntegratedGPU(hSysmanDevice)) {
+            sysmanDevicesToQuery.push_back(sysmanDevice);
+        }
+    }
+
+    env->ReleaseLongArrayElements(jSysmanDevices, sysmanDevices, 0);
+
+    jlongArray sysmanDevicesArray = env->NewLongArray(sysmanDevicesToQuery.size());
+    if (sysmanDevicesArray == NULL) {
+        return NULL; // Out of memory error
+    }
+
+    env->SetLongArrayRegion(sysmanDevicesArray, 0, sysmanDevicesToQuery.size(), sysmanDevicesToQuery.data());
+
+    return sysmanDevicesArray;
+
+}
+
+/*
+ * Class:     uk_ac_manchester_tornado_drivers_spirv_levelzero_LevelZeroPowerMonitor
+ * Method:    getEnergyCounters
+ * Signature: (J)Ljava/util/List;
+ */
+JNIEXPORT jobject JNICALL Java_uk_ac_manchester_tornado_drivers_spirv_levelzero_LevelZeroPowerMonitor_getEnergyCounters
+  (JNIEnv *env, jobject obj, jlong sysmanDeviceHandle){
+
+    zes_device_handle_t hSysmanDevice = reinterpret_cast<zes_device_handle_t>(sysmanDeviceHandle);
+
+    uint32_t numPowerDomains = 0;
+    ze_result_t result = zesDeviceEnumPowerDomains(hSysmanDevice, &numPowerDomains, nullptr);
+    if (result != ZE_RESULT_SUCCESS) {
+        std::cerr << "Error enumerating power domains." << std::endl;
+        return nullptr; 
+    }
```

Comment:
we have the `LOG_ZE_JNI` function defined in `ze_log.h`. Please reuse this function to have a unifier error report.

Reply:
Update all printed messages, accordingly.